app/src/main/java/de/westnordost/streetcomplete/data/visiblequests/VisibleQuestTypeController.kt

```diff
@@ -42,6 +42,12 @@ class VisibleQuestTypeController(
 
     private val selectedPresetId: Long get() = questPresetsSource.selectedId
 
+    fun copyVisibilities(presetId: Long, newPresetId: Long) {
+        val duplicatedQuestTypes = getVisible(presetId)
+        val questTypes = questTypeRegistry.associateWith { it in duplicatedQuestTypes }
+        setVisibilities(questTypes, newPresetId)
```

Comment:
```suggestion
        visibleQuestTypeDao.putAll(newPresetId, visibleQuestTypeDao.getAll(presetId))
        if (newPresetId == selectedPresetId) onQuestTypeVisibilitiesChanged()
```

I think I'd prefer this. It is more precise (`getVisible` does not return exactly what is set through `visibleQuestTypeDao.getAll` returns) and does not take the detour of getting the quest type classes.